src/insights_client/__init__.py

```diff
@@ -214,16 +204,26 @@ def run_phase(phase, client, validated_eggs):
     ] + sys.argv[1:]
     config = client.get_conf()
 
-    all_eggs = [ENV_EGG, NEW_EGG] + validated_eggs
+    all_eggs = [NEW_EGG] + validated_eggs
+    if ENV_EGG is not None:
+        all_eggs = [ENV_EGG] + all_eggs
+        logger.debug("ENV_EGG defined as %s." % ENV_EGG)
```

Comment:
The name of the variable is implementation detail. I'd do something like

```py
logger.debug("Using environment egg '%s'.".format(ENV_EGG))
```

instead.

Reply:
Or even better

```suggestion
        logger.debug("Using environment egg %s.", ENV_EGG)
```

The Python variable name is an implementation detail, but the environment variable name isn’t – that’s an API. But that’s EGG, not ENV_EGG and “EGG defined as %s.” sounds rather cryptic.

Still, however, even a Python variable name may make sense in a debug message if its audience is the developers. But it’s “verbose”, not “debug”, so it sounds more like for power users rather than developers. 👍🏻